Source/ARTRealtime.m

```diff
@@ -536,13 +569,26 @@ - (void)onClosed {
     }
 }
 
+- (void)onAuth {
+    [self.logger info:@"R:%p server has requested an authorise", self];
+    switch (self.connection.state) {
+        case ARTRealtimeConnecting:
+        case ARTRealtimeConnected:
+            [self transportReconnectWithRenewedToken];
+            break;
+        default:
+            NSAssert(false, @"Invalid Realtime state: expected Connecting or Connected");
```

Comment:
I would reserve those assertions for bugs _on the library_. If e. g. the server sends an AUTH while the connection is closing, that would be a bug in the server, and we probably should treat those more gracefully than crashing, e. g. log and ignore.